src/widget/wtrackmenu.cpp

```diff
@@ -138,6 +139,166 @@ void WTrackMenu::createMenus() {
+                                        .arg(primaryArtistText),
+                                [this, primaryArtist]() {
+                                    m_pLibrary->searchTracksInCollection(
+                                            QStringLiteral("artist:\"") +
+                                            primaryArtist + QChar('"'));
+                                });
+                        if (!secondaryArtist.isEmpty()) {
+                            m_pSearchRelatedMenu->addAction(
+                                    tr("Artist \"%1\"")
+                                            .arg(secondaryArtistText),
+                                    [this, secondaryArtist]() {
+                                        m_pLibrary->searchTracksInCollection(
+                                                QStringLiteral("artist:\"") +
+                                                secondaryArtist + QChar('"'));
+                                    });
+                        }
+                        m_pSearchRelatedMenu->addAction(
+                                tr("Album artist \"%1\"")
+                                        .arg(primaryArtistText),
+                                [this, primaryArtist]() {
+                                    m_pLibrary->searchTracksInCollection(
+                                            QStringLiteral("album_artist:\"") +
+                                            primaryArtist + QChar('"'));
+                                });
+                        if (!secondaryArtist.isEmpty()) {
+                            m_pSearchRelatedMenu->addAction(
+                                    tr("Album artist \"%1\"")
+                                            .arg(secondaryArtistText),
+                                    [this, secondaryArtist]() {
+                                        m_pLibrary->searchTracksInCollection(
+                                                QStringLiteral(
+                                                        "album_artist:\"") +
+                                                secondaryArtist + QChar('"'));
+                                    });
+                        }
+                    }
+                    const auto albumTitle = pTrack->getAlbum();
+                    if (!albumTitle.isEmpty()) {
+                        const auto albumTitleText =
+                                mixxx::escapeTextPropertyWithoutShortcuts(albumTitle);
+                        m_pSearchRelatedMenu->addAction(
+                                tr("Album \"%1\"")
+                                        .arg(albumTitleText),
+                                [this, albumTitle]() {
+                                    m_pLibrary->searchTracksInCollection(
+                                            QStringLiteral("album:\"") +
+                                            albumTitle + QChar('"'));
+                                });
```

Comment:
This will be really useful for adding a whole album to AutoDJ. :smile:

Reply:
The same applies for the parent folder search, depending on how you organize your music.